### DISCUSSION TITLE
CUDA constant memory

### DISCUSSION THREAD

**Discussion by jinz2014** [intel/llvm#15815] [Q&A]
Does SYCL support CUDA constant memory ?

**Comment by AlexeySachkov** [comment]
Hi @jinz2014,

In general, SYCL has no notion of constant address space - it was removed in SYCL 2020 for simplicity reasons, I believe. Therefore, the remaining question is whether or not the compiler is able to optimize certain things to use backend-specific features.

I'm not familiar with the CUDA, so I will tag @jchlanda here to help with a more detailed answer

**Comment by jchlanda** [comment]
@JackAKirk has looked into this area recently.

**Comment by JackAKirk** [comment]
> Does SYCL support CUDA constant memory ?

Yes this is going to be available very soon via `device_global`. Probably by just pre-pending `device_global` by `const` but I am finalizing the details.

**Comment by zjin-lcf** [comment]
Is device_global finalized for CUDA constant memory ?
